coptic_char_generator.py

```diff
@@ -147,8 +61,12 @@ def train_batch(model, optimizer, criterion, data, data_indexes, update=True):
 
         index_tensor = torch.tensor(data_item.indexes, dtype=torch.int64).to(device)
         label_tensor = torch.tensor(data_item.labels, dtype=torch.int64).to(device)
-        out, hidden = model([index_tensor])  # [:-1]
-        loss = criterion(out[0], label_tensor)  # [1:]
+        print(label_tensor)
+        out = model([index_tensor])  # [:-1]
+        print(f"out: {out.view(-1, out.shape[-1]).shape}")
+        print(f"out[0]: {out.view(-1, out.shape[-1]).shape}")
+        print(f"label: {label_tensor.view(-1).shape}")
+        loss = criterion(out.view(-1, out.shape[-1]), label_tensor.view(-1))  # [1:]
```

Comment:
spent a lot of time trying different things but haven't made much progress. gonna pick up on looking at the loss section shortly, maybe switch it over to accuracy and see. labels are supposed to make it so that we're only evaluating the masked tokens, but i'm not really understanding how that is happening